drivers/hud_apr/app/models/hud_apr/generators/ce_apr/fy2021/question_concern.rb

```diff
@@ -8,26 +8,56 @@ module HudApr::Generators::CeApr::Fy2021::QuestionConcern
   extend ActiveSupport::Concern
 
   included do
+    # By default, only include members with assessments
+    def self.filter_universe_members(associations)
+      associations.reject { |_, row| row[:ce_assessment_date].blank? }
+    end
+
     private def clients_with_enrollments(batch)
-      enrollment_scope.
-        joins(:project).
+      client_ids = batch.map(&:id)
+      assessed_clients = enrollment_scope.
+        joins(:project, enrollment: :assessments).
         merge(GrdaWarehouse::Hud::Project.coc_funded).
-        where(client_id: batch.map(&:id)).
+        where(client_id: client_ids).
         order(as_t[:AssessmentDate].asc).
         group_by(&:client_id).
         reject { |_, enrollments| nbn_with_no_service?(enrollments.last) }
```

Comment:
This should reject any enrollments `nbn_with_no_service?(enrollment)` and then reject the client entirely only if no enrollments remain.